### PR TITLE
Decorator-pull-requests

### PR DESCRIPTION
**Pull request by ikoote1**
Improvements made to the code:
1. Created a class Nameable.
- Implemented a method called correct_name that will raise a NotImplementedError.
2. Turned my Person class to Nameable and;

- Made sure that your Person class inherits from Nameable
- Made sure that this class has a method correct_name implemented. It should simply return the name attribute.

3. Prepared base Decorator and;

- Made sure that it inherits from Nameable.
- In the constructor, I assigned a nameable object from params to an instance variable.
- Implemented the correct_name method that returns the result of the correct_name method of the @nameable.

 4. Prepare CapitalizeDecorator and TrimmerDecorator

- For the CapitalizeDecorator:
Created a class that inherits from the base Decorator class.
Implemented a method correct_name that capitalizes the output of @nameable.correct_name.
- For the TrimmerDecorator:
Created a class that inherits from the base Decorator class.
Implemented a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

5. Opened a pull request.

Thanks for reviewing my code.

### CODE REVIEW

File: modules/person.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    trimmed_name = @nameable.correct_name[0..9]
+    trimmed_name.length < @nameable.correct_name.length ? trimmed_name : @nameable.correct_name
+  end
+end
+
```
Comment (DammyShittu) [inline modules/person.rb]: - [x] These classes do not belong here. This file should strictly be for the Person class. Kindly create new files for each class and move them to their respective files.

File: modules/person.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
+require './nameable'
```
Comment (DammyShittu) [inline modules/person.rb]: - [x] Kindly change this to `require_relative` as that is the convention for importing local files in ruby. I was unable to run your app due to an error from this. The `require` is reserved for external files like gems.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @ikoote1,

## Highlights
✔️ Class decorator feature

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @ikoote1 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Github flow correctly implemented
✔️  All functionalities implemented correctly


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**